specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json

```diff
@@ -0,0 +1,4101 @@
+            "$ref": "./examples/galleryExamples/GalleryApplicationVersion_ListByGalleryApplication.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}/share": {
+      "post": {
+        "tags": [
+          "GallerySharingUpdate"
+        ],
+        "operationId": "GallerySharingProfile_Update",
+        "description": "Update sharing profile of a gallery.",
+        "parameters": [
+          {
+            "$ref": "../../../common-types/v1/common.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "galleryName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Shared Image Gallery."
+          },
+          {
+            "$ref": "../../../common-types/v1/common.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "sharingUpdate",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SharingUpdate"
+            },
+            "description": "Parameters supplied to the update gallery sharing profile."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SharingUpdate"
+            }
+          },
+          "202": {
+            "description": "Accepted",
+            "schema": {
+              "$ref": "#/definitions/SharingUpdate"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common-types/v1/common.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Add sharing id to the sharing profile of a gallery.": {
+            "$ref": "./examples/galleryExamples/Gallery_AddToSharingProfile.json"
+          },
+          "share a gallery to community.": {
+            "$ref": "./examples/galleryExamples/Gallery_EnableCommunityGallery.json"
+          },
+          "reset sharing profile of a gallery.": {
+            "$ref": "./examples/galleryExamples/Gallery_ResetSharingProfile.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}/inVMAccessControlProfiles/{inVMAccessControlProfileName}": {
+      "put": {
+        "tags": [
+          "GalleryInVMAccessControlProfiles"
+        ],
+        "operationId": "GalleryInVMAccessControlProfiles_CreateOrUpdate",
```

Comment:
This always confused me. Why is there a CreateOrUpdate operation, then a separate Update operation? Does the CreateOrUpdate actually do an Update? Do these operations do something different or under different circumstances?

Reply:
> Does the CreateOrUpdate actually do an Update?

Yes

> Do these operations do something different or under different circumstances?

No. Both operations are handled in the exact same manner in the service side. Both operations follow patch semantics, and this is the usual pattern for all our resources. I believe it's the same with other RPs in Compute.